pkg/actuators/machine/machine_scope.go

```diff
@@ -33,6 +33,8 @@ type machineScopeParams struct {
 	machine *machinev1beta1.Machine
 	// api server controller runtime client for the openshift-config-managed namespace
 	configManagedClient runtimeclient.Client
+	// accessKeyID (string) to *DescribeRegionsData map
```

Comment:
Nice, that should mean that if the credentials change we get a new cache, which makes sense. Good idea!

Reply:
Of course, it was your idea :D
https://github.com/openshift/machine-api-provider-aws/pull/32/files#r848333870